src/java.base/share/classes/java/net/HostPortrange.java

```diff
@@ -60,6 +60,11 @@ public int hashCode() {
     }
 
     HostPortrange(String scheme, String host) {
+        // Defensive validation first
+        if (host == null || host.isEmpty()) {
+            throw new IllegalArgumentException("empty authority");
```

Comment:
Hello Oumaiyma, the change overall looks OK to me. However, this error message and the other one feel a bit odd. The variable is called `host` and we throw an "empty authority" here whereas the other place, the variable is called "authority" and we throw a error message which says "empty host". 

@dfuch do you suggest we switch these error messages?